src/identity/components/Identity.tsx

```diff
@@ -14,10 +15,15 @@ export function Identity({
 }: IdentityReact) {
   const { chain: contextChain } = useOnchainKit();
   const accountChain = chain ?? contextChain;
+  const { address } = useAccount();
+
+  if (!address && !addressProp) {
+    return null;
+  }
 
   return (
     <IdentityProvider
-      address={address}
+      address={addressProp || address}
```

Comment:
if you did the above, you wouldn't need this conditional, you could just use the return value from the intercepter hook